clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -122,9 +126,27 @@ else if (clientAuthConfig.equals("requested"))
         }
     }
 
+    /*
+    Method to load third party java security provider.
+    ** Not equivalent to SSL.Provider config which expects a full ssl protocol implementation
+     */
+    private void loadThirdPartyProviders(){
+        if(appkiProvider != null){
+            try {
+                Security.addProvider((Provider) Class.forName(appkiProvider).newInstance());
+            } catch (InstantiationException e) {
+                throw new KafkaException("Could not instantiate AP PKI provider "+ appkiProvider, e);
+            } catch (IllegalAccessException e) {
+                throw new KafkaException(e);
+            } catch (ClassNotFoundException e) {
+                throw new KafkaException("Could not find AP PKI provider "+ appkiProvider, e);
+            }
+        }
+    }
 
     private SSLContext createSSLContext() throws GeneralSecurityException, IOException  {
         SSLContext sslContext;
+        loadThirdPartyProviders();
```

Comment:
For getting SSLContext instance (line 150 - 153), do you need to use your APPKIProviver for it?